library/network_connections.py

```diff
@@ -2511,6 +2513,11 @@ def run_prepare(self):
                     "unsupported type %s for initscripts provider"
                     % (connection["type"]),
                 )
+            if not connection["interface_name"] and connection["ip"]["route"]:
+                self.log_warn(
+                    idx,
+                    "the route will be configured without the 'dev' attribute",
+                )
```

Comment:
What's the reason to put the warning here instead of in an else clause above? Also, the message needs to be more specific so it is clear which route is meant and @ffmancera proposed some different wording to explain the behavior.

Reply:
I put it here because I found out almost all the loggings for the initscripts provider was put in the `Cmd_initscripts` class rather than  `IfcfgUtil.ifcfg_create()` function.

Reply:
The new warning message is fine for me :-)